src/main/java/org/opensearch/flowframework/common/FlowFrameworkSettings.java

```diff
@@ -44,4 +44,14 @@ private FlowFrameworkSettings() {}
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
+
+    /** This setting sets the maximum number of transport request retries */
+    public static final Setting<Integer> MAX_REQUEST_RETRY = Setting.intSetting(
+        "plugins.flow_framework.max_request_retry",
+        5,
+        0,
+        20,
```

Comment:
Did any plugin have the same default and max values for retrying?

Reply:
I have not seen a similar setting from a different plugin. However within the neural plugin, their maximum retry is set to 3 [here](https://github.com/opensearch-project/neural-search/blob/b3c73bdaae28629d9be4128afcf549c790070887/src/main/java/org/opensearch/neuralsearch/util/RetryUtil.java#L18). 

It's important to sleep the thread between retry attempts, I've observed setting 100+ retries would still not be enough without having this wait time. I think a default of 5 retries is reasonable, since model registration takes some time, and limiting the maximum number of retries and forcing additional workflow steps to wait rather than continuously send transport requests is a better way to go about this. 

I'm open to suggestions on what to set the maximum value to

Reply:
I don't have any strong opinion here. @dbwiddis thoughts?

Reply:
I'm OK with a very large max, or even no max value (I think if you just specify min, the max is `Integer.MAX_VALUE`) as long as we have a sane default.